tests/attribute_tests.cpp

```diff
@@ -0,0 +1,58 @@
+#define BOOST_TEST_MODULE RadProtoAttributeTests
+
+#include "radproto/attribute.h"
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#pragma GCC diagnostic ignored "-Wsign-compare"
+#pragma GCC diagnostic ignored "-Wparentheses"
+#include <boost/test/unit_test.hpp>
+#pragma GCC diagnostic pop
+
+BOOST_AUTO_TEST_SUITE(RadProtoAttributeTests)
+
+BOOST_AUTO_TEST_CASE(ClassStringDataConstructor)
+{
+    std::vector<uint8_t> d {'t', 'e', 's', 't'};
+    RadProto::String s(1, d.data(), d.size());
+
+    BOOST_CHECK_EQUAL(s.toString(), "test");
+
+    std::vector<uint8_t> values = s.toVector({}, {});
+    std::vector<uint8_t> expected {1, 6, 't', 'e', 's', 't'};
+
+    BOOST_CHECK_EQUAL_COLLECTIONS(values.begin(), values.end(), expected.begin(), expected.end());
```

Comment:
Nice finding! But it is a little bit too old-fashioned. Let's use modern alternative:
```
BOOST_TEST(values == expected, boost::test_tools::per_element());
```
Details: https://www.boost.org/doc/libs/1_85_0/libs/test/doc/html/boost_test/utf_reference/testing_tool_ref/assertion_boost_test_universal_macro.html

Reply:
Same below.